### PR TITLE
feat: Balance & lock

### PR DESCRIPTION
**Pull request by EliasEriksson**
* added interface for Balance
* added interface for Lock

### CODE REVIEW

File: Balance/Lock/index.spec.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { weekmeter } from "../../index"
+
+describe("Lock", () => {
+	it("is", () => {
+		const lock: weekmeter.Balance.Lock = { date: "2023-01-01" }
+		expect(weekmeter.Balance.Lock.is(lock)).toEqual(true)
+		expect(weekmeter.Balance.Lock.is((({ date, ...lock }) => lock)(lock))).toEqual(false)
+		// expect(weekmeter.Balance.Lock.is((({ user, ...lock }) => lock)(lock))).toEqual(false)
```
Comment (TheLillyleaf) [inline Balance/Lock/index.spec.ts]: comment

Comment (LinusDahlstrom) [inline Balance/Lock/index.spec.ts]: why is this a comment?

File: Balance/index.spec.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { weekmeter } from "../index"
+
+describe("Balance", () => {
+	it("is", () => {
+		const balance: weekmeter.Balance = { value: { hours: 1, minutes: 30 }, lock: "2023-01-01" }
+		expect(weekmeter.Balance.is(balance)).toEqual(true)
+		expect(weekmeter.Balance.is((({ lock, ...balance }) => balance)(balance))).toEqual(true)
+		expect(weekmeter.Balance.is((({ value, ...balance }) => balance)(balance))).toEqual(false)
+	})
+})
```
Comment (LinusDahlstrom) [inline Balance/index.spec.ts:10]: should we have a test where we try to change a balance that has been locked?

File: Time/index.ts [hunk 3]
```diff
@@ -7,7 +7,7 @@ import { Changeable as TimeChangeable } from "./Changeable"
 import { Creatable as TimeCreatable } from "./Creatable"
 
 export interface Time extends Time.Creatable {
-	locked?: true
+	locked?: { expected: isoly.TimeSpan }
```
Comment (LinusDahlstrom) [inline Time/index.ts:10]: what is the thought behind calling the property `expected` ?

**Review by LinusDahlstrom — CHANGES_REQUESTED** [review]

**Review by TheLillyleaf — CHANGES_REQUESTED** [review]

**Review by TheLillyleaf — APPROVED** [review]

**Review by LinusDahlstrom — APPROVED** [review]

**Pull request merged**